### PR TITLE
Added migration to store trial period days on tier

### PR DESCRIPTION
**Pull request by rshbhgrg**
refs https://github.com/TryGhost/Team/issues/1724

- storing trial period days allows site owners to give free trial for N days to all members signing up on a tier
- the new column name `trial_period_days` is consistent with naming used by Stripe for passing trial value
- by default, all tiers have 0 or no trial period days

### CODE REVIEW

File: ghost/core/core/server/data/schema/schema.js [hunk 1]
```diff
@@ -422,6 +422,7 @@ module.exports = {
             defaultTo: 'none',
             validations: {isIn: [['public', 'none']]}
         },
+        trial_period_days: {type: 'integer', unsigned: true, nullable: false, defaultTo: 0},
```
Comment (rishabhgrg) [inline ghost/core/core/server/data/schema/schema.js:425]: The naming `trial_period_days` matches naming used by Stripe for passing trial days value for subscriptions/sessions.

**Review by daniellockyer — APPROVED** [review]

File: ghost/core/core/server/data/schema/schema.js (hunk 1 above)
Comment (ErisDS) [reply ghost/core/core/server/data/schema/schema.js:425]: @allouis qq: what do you think about matching Stripe here? I am on the fence.

File: ghost/core/test/e2e-api/admin/__snapshots__/members-newsletters.test.js.snap [hunk 2]
```diff
@@ -63,7 +63,7 @@ exports[`Members API - With Newsletters - compat mode Can fetch members who are
 Object {
   "access-control-allow-origin": "http://127.0.0.1:2369",
   "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
-  "content-length": "2056",
```
Comment (ErisDS) [inline ghost/core/test/e2e-api/admin/__snapshots__/members-newsletters.test.js.snap:66]: How to tell there is no mapper in place 😂

**Review by ErisDS — APPROVED** [review]
Looks good to me. Have asked for Egg's opinion on naming just to check in with his current thinking.

File: ghost/core/test/e2e-api/admin/__snapshots__/members-newsletters.test.js.snap (hunk 2 above)
Comment (rishabhgrg) [reply ghost/core/test/e2e-api/admin/__snapshots__/members-newsletters.test.js.snap:66]: 😄 Yeah that bugged me as well. I have marked it in my todos to have a look towards the end of Free trials if time permits to sort out

File: ghost/core/core/server/data/schema/schema.js (hunk 1 above)
Comment (allouis) [reply ghost/core/core/server/data/schema/schema.js:425]: Given that this table is for storing Tiers which are a Ghost concept, there's no reason to match Stripe naming IMO.

However, I don't think the name is necessarily bad, but if there is a better name we should go for it!

All I can think is `trial_period` or `trial_days` which is a lil shorter 🦐

Comment (rishabhgrg) [reply ghost/core/core/server/data/schema/schema.js:425]: @ErisDS Do you have any pref on the naming between these options ? :)

Comment (ErisDS) [reply ghost/core/core/server/data/schema/schema.js:425]: `trial_days` feels nicest to me

Comment (rishabhgrg) [reply ghost/core/core/server/data/schema/schema.js:425]: 👌 Yeah sounds good to me too, will update, thanks.

**Review by daniellockyer — APPROVED** [review]

**Pull request merged**